lib/mv_opentelemetry/dataloader.ex

```diff
@@ -27,23 +28,55 @@ defmodule MvOpentelemetry.Dataloader do
     :ok
   end
 
-  def handle_event([:dataloader, :source, :batch, :run, :start], _measurements, meta, opts) do
+  def handle_event(
+        [:dataloader, :source, :batch, :run, :stop],
+        measurements,
+        %{batch: batch},
+        opts
+      ) do
     event_name = "dataloader.source.batch.run"
 
-    parent_context = OpentelemetryProcessPropagator.fetch_parent_ctx(1, :"$callers")
-    attach_context(parent_context)
+    batch_attributes =
+      case batch do
+        {{:queryable, _pid, queryable, cardinality, col, _}, _} ->
+          [
+            {"dataloader.source.batch.type", "queryable"},
+            {"dataloader.source.batch.queryable", queryable},
+            {"dataloader.source.batch.cardinality", cardinality},
+            {"dataloader.source.batch.column", col}
+          ]
 
-    OpentelemetryTelemetry.start_telemetry_span(__MODULE__, event_name, meta, %{})
-    |> Span.set_attributes(opts[:default_attributes])
+        {{:assoc, schema, _pid, assoc_field, queryable, _}, _} ->
+          [
+            {"dataloader.source.batch.type", "assoc"},
+            {"dataloader.source.batch.schema", schema},
+            {"dataloader.source.batch.assoc_field", assoc_field},
+            {"dataloader.source.batch.queryable", queryable}
+          ]
 
-    :ok
-  end
+        _ ->
+          []
+      end
+
+    %{duration: duration} = measurements
+    end_time = :opentelemetry.timestamp()
+    start_time = end_time - duration
+    attributes = batch_attributes ++ opts[:default_attributes]
+    span_opts = %{start_time: start_time, attributes: attributes}
```

Comment:
Like with ecto we need to cheat here, dataloader is not actually emitting span events, just an event at the beginning and at the end.